### DISCUSSION TITLE
我能将此模板上传到 Overleaf Gallery 里吗？

### DISCUSSION THREAD

**Discussion by framist** [note286/xduts#114] [Ideas]
*Can I publishing this template into the Overleaf Gallery?*

[Overleaf Gallery](https://www.overleaf.com/latex/templates) 是一个可公开访问的模板和示例集合。

为了照顾众多像我一样的 $\LaTeX$ 小白，我想将您的模板稍作包装——比如增加一个 `.tex`  文件以供预览，因为一个可以编译的文件是必要的——传到 Overleaf Gallery 里，方便更多人使用。虽然这种方式不太优雅，但是也可以让更多人使用 $\LaTeX$ ，使用 xduts。

---

参考：

[如何在 Overleaf Gallery 发布模板-官方文档](https://www.overleaf.com/learn/how-to/Publishing_templates_and_examples_in_the_Overleaf_Gallery)

**Comment by sikouhjw** [comment]
1. https://github.com/note286/xduts/blob/3237abfaf40e7c94a8e2b200bd4f5b2cb9b2c8bf/README.md?plain=1#L50-L52
2. 我建议 @note286 在 Releases 里发布编译好、可直接使用的模板，方便上传
3. 上传到 Overleaf 应该让模板作者上传

**Reply by note286** [reply]
Releases我尝试过一次，我懒得去弄，没有啥必要，通过阅读README用户可以自动编译出文档类文件和手册文件，如果这一步卡住了，后续问题也不断。

虽然Overleaf用户无法本地编译，但是我在README中提供了TeX Live编译好的链接，直接下载即可。我更新CTAN还是很及时的，所以这一点问题不大。

**Reply by framist** [reply]
考虑到 LPPL-1.3c license 。非模板作者上传应该也是自由的吧

> 根据 LPPL-1.3c 许可证的规定，作者授予了使用该许可证下软件的用户一些特定的权利，这些权利包括使用、修改和分发该软件。因此，只要您遵守该许可证的要求和限制，您就有权在其他平台上发布该软件或该软件的修改版本，而作者无权拒绝您这样做。

**Reply by framist** [reply]
关于*Release 里发布编译好、可直接使用的模板*， @note286 也提过：
> XDUTS的设计就是只要你能走到编译那一步，你就能编译出PDF，你就可以开始写你论文的内容了，至于我提供的各种功能，看不懂就不用，没关系。最直接的原因就是我不想把这个项目弄得不伦不类，我希望README和手册里介绍的都是和XDUTS直接相关的内容，而不是参杂着LaTeX入门教程。

**Reply by note286** [reply]
而且，我并不想维护多个，很麻烦，我只需要保证GitHub的更新与维护即可。

**Reply by note286** [reply]
@framist 你也可以说一下你希望示例文件里有什么？

**Reply by framist** [reply]
我就准备只写个“不会使用此模板？快去阅读 XDUTS 手册！”😂，毕竟 Overleaf Gallery  要求必须有个过编译的 pdf
如果其他人有需求，我可能会加入更多内容。

**Reply by note286** [reply]
我不知道你调研过多少西安电子科技大学的LaTeX用户。我目前的结论就是Overleaf暂时没有任何必要去添加一个示例文件，为了弥补广大西安电子科技大学学子的LaTeX知识水平的欠缺，我给了 https://github.com/note286/xduts/discussions/40 ，这个你应该看见了，我觉得已经足够了。

**Reply by note286** [reply]
@framist 我知道你的出发点是好的，希望有更多同学更易上手，非常感谢你的支持。但是，一个示例文件并不能解决这个问题，几乎没有任何帮助，我解释一下为什么？如果xduts有示例文件，会包含两部分，一部分就是常见的代码片段，例如子图、子表、算法、公式、跨页表格，等等。对于这些，每个用户都有自己熟悉的宏包，一般来说，这些宏包都是不兼容的，如果你想塞进去这些，你要准备多少个示例文件，如果只给出一个，给出的不是用户会用的，他还要删除掉，没有意义。还有些用户是第一次用LaTeX，没有熟悉的宏包，这些用户即使你给出一个示例，也不见得就学会了，还是要多看多练，那么给出的这么一个示例，几乎没有帮助。对此， https://github.com/note286/xduts/discussions/40 部分解决了这个问题，另外我还在手册中给出了推荐宏包，只要有点学习能力的人，哪怕是去百度搜索，都能学会。而且我还开篇就提到了先看lshort。另外一部分就是和xduts紧密相关的，例如各种功能开关，信息配置等，这部分即使给出了，用户仍需要去查看手册才明白什么意思，如果把手册中的内容搬过去，那么手册就没必要存在了。而且，PDF的展示效果要比文本丰富的多。

最后说几句，我接触了太多LaTeX用户，当然我的观察是片面的，样本是不足的，结论可能也是不对的。对于有一定学习能力的人，xduts目前的状态就挺好，没有任何问题，作为开发者的我，维护起来也容易。对于没有学习能力的人，你就算喂到嘴边，都懒得嚼，除非你给写好，但是，我只是一个开源项目的开发者，只是爱好而已，我没有拯救ta的义务啊。这就是现实，没有办法。

**Reply by framist** [reply]
我其实也认为完全是足足足够了。我做这事是应一些同学的需求。毕竟对于部分特别是非计算机专业的同学，入门再使用比先使用再入门难得多。

**Reply by note286** [reply]
@framist 我非常理解，但是现实就是LaTeX的使用不是一个示例文件能解决的，对此几乎没有任何帮助，如果用户没有一定的学习能力，还不如使用Microsoft Word，花精力在内容上。

你可以观察一下这些同学，是否尝试去了解一下LaTeX，还是只是听说LaTeX排版效果好，就想试试，结果README都没读完，就说不会用。很多用户手册也不怎么看，甚至都不会Ctrl+F，例如想要删除页眉，Ctrl+F页眉就好，但是就是不搜，你有啥办法。

我也和不少xduts的用户聊过，自己也思考过，最后的结论就是就这么着吧，我又不是党和政府，有责任照顾好每一个人，爱用不用吧。用户自己都不上心，你上赶着去喂没有任何意义。

**Reply by framist** [reply]
嗯，感谢理解！我明白了

**Reply by note286** [reply]
@framist 我的内测群里也有非计算机专业的用户，我问过，都没什么问题。例如，用户不知道CMD是啥，百度一搜不就知道了。经常遇到最离谱的就是很多用户看见一个不认识的词语或者概念，不知道去搜索，2023年了，可悲。

**Comment by note286** [comment]
非常抱歉，暂时不考虑提交至Overleaf Gallery，原因如下：

1. 由于xduts尚未稳定，今年会有一个较大变动的改版。
2. 另外，xduts是一个充满个人风格的作品，我目前对xduts提供示例文件是排斥的，xduts的设计是十分简约的，和使用标准文档类一样，不需要任何示例文件，直接指定文档类即可编译。使用xduts独有的接口，用户阅读手册即可，示例文件无非是将`.pdf`转为`.tex`而已，没有任何意义，而且阅读体验大打折扣。

**Reply by sikouhjw** [reply]
>  使用xduts独有的接口，用户阅读手册即可，示例文件无非是将`.pdf`转为`.tex`而已，没有任何意义，而且阅读体验大打折扣。

这点我的意见跟你不同，对于推广的层面，一个直观的 `.tex` 还是很重要的……

**Reply by note286** [reply]
你说的没错，一般用户还是非常喜欢一个可以编辑的示例文件。至少对于xduts，我认为完全没有必要，不可否认的是，手册中的内容是一定要看的，在示例文件中再给一遍没有什么意义。如果用户卡在这一步，后续还会有更多的麻烦，还不如Microsoft Word。

或者，等哪天我想通的吧。
